### PR TITLE
Update OpenAI and Azure OpenAI connector setup guides

### LINKED ISSUE #5357: [Request] Updates to OpenAI connector setup guide
### Description

Charles gave some feedback on the doc, this issue tracks updating it with his requested changes.

### Background & resources

* PRs: 
* Issues/metas: 
* Point of contact: 
* Test environments: 

### Which documentation set does this change impact?

ESS and serverless

### ESS release

8.14

### Serverless release

-

### Feature differences

-

### API docs impact

-

### Prerequisites, privileges, feature flags

_No response_

### PR DESCRIPTION
**Pull request by benironside**
Fixes #5357 by making some minor updates to the OpenAI connector setup guide and the Azure OpenAI setup guide.

Previews: [connect to Azure OpenAI](https://docs-elastic-9676w7w6c-elastic-dev.vercel.app/serverless/security/connect-to-azure-openai)
[Connect to OpenAI](https://docs-elastic-9676w7w6c-elastic-dev.vercel.app/serverless/security/connect-to-openai)

### CODE REVIEW

**Comment by elasticdocs** [comment]
:rocket: Built _co-preview-docs_ successfully!

* https://docs-elastic-9676w7w6c-elastic-dev.vercel.app

* Built with commit bd44bab8eb61f165af548cc2e8b27903d6eff178

> Issues? Visit #next-docs in Slack

**Review by Charelzard — APPROVED** [review]
Approved for both OpenAI and Azure OpenAI

File: docs/assistant/connect-to-openai.asciidoc [hunk 1]
```diff
@@ -12,7 +12,7 @@ This page provides step-by-step instructions for setting up an OpenAI connector
 
 Before creating an API key, you must choose a model. Refer to the https://platform.openai.com/docs/models/gpt-4-turbo-and-gpt-4[OpenAI docs] to select a model. Take note of the specific model name (for example `gpt-4-turbo`); you'll need it when configuring {kib}.
 
-NOTE: `GPT-4 Turbo` offers increased performance. `GPT-4` and `GPT-3.5` are also supported. 
+NOTE: GPT-4o offers increased performance over previous versions. For more information on how different models perform for different tasks, refer to the <<llm-performance-matrix>>.
```
Comment (nastasha-solomon) [inline docs/assistant/connect-to-openai.asciidoc:15]: Feel free to ignore if you didn't intend to style GPT-4o here. :)

```suggestion
NOTE: `GPT-4o` offers increased performance over previous versions. For more information on how different models perform for different tasks, refer to the <<llm-performance-matrix>>.
```

File: docs/assistant/azure-openai-setup.asciidoc [hunk 2]
```diff
@@ -72,7 +72,7 @@ Now, set up the Azure OpenAI model:
 ** If you select `gpt-4`, set the **Model version** to `0125-Preview`. 
 ** If you select `gpt-4-32k`, set the **Model version** to `default`.
 +
-IMPORTANT: The models available to you will depend on https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models#model-summary-table-and-region-availability[region availability]. For best results, use `GPT 4 Turbo version 0125-preview` or `GPT 4-32k` with the maximum Tokens-Per-Minute (TPM) capacity. In most regions, the GPT 4 Turbo model offers the largest supported context window.
+IMPORTANT: The models available to you depend on https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models#model-summary-table-and-region-availability[region availability]. For best results, use GPT-4o 2024-05-13 with the maximum Tokens-Per-Minute (TPM) capacity. For more information on how different models perform for different tasks, refer to the <<llm-performance-matrix>>.
```
Comment (nastasha-solomon) [inline docs/assistant/azure-openai-setup.asciidoc:75]:
```suggestion
IMPORTANT: The models available to you depend on https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models#model-summary-table-and-region-availability[region availability]. For best results, use `GPT-4o 2024-05-13` with the maximum Tokens-Per-Minute (TPM) capacity. For more information on how different models perform for different tasks, refer to the <<llm-performance-matrix>>.
```

File: docs/serverless/assistant/connect-to-azure-openai.mdx [hunk 3]
```diff
@@ -56,7 +56,7 @@ Now, set up the Azure OpenAI model:
 8. Click **Create**.
 
 <DocCallOut title="Important" color="warning">
-The models available to you will depend on [region availability](https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models#model-summary-table-and-region-availability). For best results, use `GPT 4 Turbo version 0125-preview` or `GPT 4-32k` with the maximum Tokens-Per-Minute (TPM) capacity. In most regions, the GPT 4 Turbo model offers the largest supported context window.
+The models available to you will depend on [region availability](https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models#model-summary-table-and-region-availability). For best results, use GPT-4o 2024-05-13 with the maximum Tokens-Per-Minute (TPM) capacity. For more information on how different models perform for different tasks, refer to the <DocLink id="llm-performance-matrix" text="LLM performance matrix"/>.
```
Comment (nastasha-solomon) [inline docs/serverless/assistant/connect-to-azure-openai.mdx:59]:
```suggestion
The models available to you will depend on [region availability](https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models#model-summary-table-and-region-availability). For best results, use `GPT-4o 2024-05-13` with the maximum Tokens-Per-Minute (TPM) capacity. For more information on how different models perform for different tasks, refer to the <DocLink id="llm-performance-matrix" text="LLM performance matrix"/>.
```

File: docs/serverless/assistant/connect-to-openai.mdx [hunk 4]
```diff
@@ -18,7 +18,7 @@ This page provides step-by-step instructions for setting up an OpenAI connector
 Before creating an API key, you must choose a model. Refer to the [OpenAI docs](https://platform.openai.com/docs/models/gpt-4-turbo-and-gpt-4) to select a model. Take note of the specific model name (for example `gpt-4-turbo`); you'll need it when configuring ((kib)).
 
 <DocCallOut title="Note">
-`GPT-4 Turbo` offers increased performance. `GPT-4` and `GPT-3.5` are also supported.
+NOTE: GPT-4o offers increased performance over previous versions. For more information on how different models perform for different tasks, refer to the <DocLink id="llm-performance-matrix" text="LLM performance matrix"/>.
```
Comment (nastasha-solomon) [inline docs/serverless/assistant/connect-to-openai.mdx:21]:
```suggestion
`GPT-4o` offers increased performance over previous versions. For more information on how different models perform for different tasks, refer to the <DocLink id="llm-performance-matrix" text="LLM performance matrix"/>.
```

**Review by nastasha-solomon — DISMISSED** [review]
Left some styling suggestions and fixed a minor typo. LGTM overall, tho!

**Review by nastasha-solomon — APPROVED** [review]

**Pull request merged**